tests/config/write.c

```diff
@@ -670,7 +670,6 @@ void test_config_write__locking(void)
 	git_transaction_free(tx);
 
 	/* Now that we've unlocked it, we should see both updates */
-	cl_git_pass(git_config_open_ondisk(&cfg, filename));
```

Comment:
I added a separate block beneath. That way, we can check the changes through the `git_config` that was part of the transaction, and then test that the changes were really persisted.